### PR TITLE
Add a new `.literal` argument

### LINKED ISSUE #158: `glue_col` cannot escape certain characters ``{ ' " `}``; fails to print to console
Hi Jim,

I came across a bug wherein certain symbols ``{ '  " ` }`` cannot be escaped in `glue_col` to print colored output and I wanted to make a note of it. I couldn't find any documentation on special escaping practices in `glue`, nor [python fstring examples](https://realpython.com/python-f-strings/#quotation-marks).

Many thanks for your attention!

### reprex
```r
library(glue)
packageDescription("glue")

# Package: glue
# Title: Interpreted String Literals
# Version: 1.3.0
# Authors@R: person("Jim", "Hester", email =
#                     "james.f.hester@gmail.com", role = c("aut", "cre"))
# Description: An implementation of interpreted string literals,
# inspired by Python's Literal String Interpolation
#        <https://www.python.org/dev/peps/pep-0498/> and Docstrings
#        <https://www.python.org/dev/peps/pep-0257/> and Julia's
# Triple-Quoted String Literals
# <https://docs.julialang.org/en/stable/manual/strings/#triple-quoted-string-literals>.
#   Depends: R (>= 3.1)
# Imports: methods
# Suggests: testthat, covr, magrittr, crayon, knitr, rmarkdown, DBI,
# RSQLite, R.utils, forcats, microbenchmark, rprintf, stringr,
# ggplot2, dplyr, withr
# License: MIT + file LICENSE
# Encoding: UTF-8
# LazyData: true
# RoxygenNote: 6.0.1
# URL: https://github.com/tidyverse/glue
# BugReports: https://github.com/tidyverse/glue/issues
# VignetteBuilder: knitr
# ByteCompile: true
# NeedsCompilation: yes
# Packaged: 2018-07-17 14:19:58 UTC; jhester
# Author: Jim Hester [aut, cre]
# Maintainer: Jim Hester <james.f.hester@gmail.com>
#   Repository: CRAN
# Date/Publication: 2018-07-17 19:10:02 UTC
# Built: R 3.5.2; x86_64-w64-mingw32; 2019-02-17 19:37:54 UTC; windows
#
# -- File: C:/Users/a131485/Documents/R/win-library/3.5.1/glue/Meta/package.rds

sessionInfo()

# R version 3.5.1 (2018-07-02)
# Platform: x86_64-w64-mingw32/x64 (64-bit)
# Running under: Windows 10 x64 (build 17763)
#
# Matrix products: default
#
# locale:
#   [1] LC_COLLATE=English_United States.1252
# [2] LC_CTYPE=English_United States.1252
# [3] LC_MONETARY=English_United States.1252
# [4] LC_NUMERIC=C
# [5] LC_TIME=English_United States.1252
#
# attached base packages:
#   [1] stats     graphics  grDevices utils     datasets  methods   base
#
# other attached packages:
#   [1] glue_1.3.0        crayon_1.3.4      data.table_1.12.0 janitor_1.2.0
# [5] purrr_0.3.1       assertthat_0.2.0  dplyr_0.8.3       odbc_1.1.6
# [9] vroom_1.0.2
#
# loaded via a namespace (and not attached):
#   [1] Rcpp_1.0.2       rstudioapi_0.10  magrittr_1.5     hms_0.4.2
# [5] tidyselect_0.2.5 bit_1.1-14       R6_2.4.0         rlang_0.4.0
# [9] stringr_1.4.0    blob_1.1.1       tools_3.5.1      packrat_0.5.0
# [13] DBI_1.0.0        bit64_0.9-7      tibble_2.1.3     snakecase_0.11.0
# [17] stringi_1.3.1    compiler_3.5.1   pillar_1.3.1     pkgconfig_2.0.2

# Prints nothing
glue_col("{green test's}")
glue_col("{green test\'s}")
glue_col("{green test\\'s}")
glue_col("{green test\\\\'s}")
glue_col("{green test\"s}")
glue_col("{green test`s}")
glue_col("<<green test's>>", .open = '<<', .close = '>>')
# Prints as expected
glue_col("{green tests}")
glue_col("{green test]s}")
glue_col("{green test,s}")
glue_col("{green test.s}")
glue_col("{green test?s}")
glue_col("{green test!s}")
glue_col("{green test-s}")
glue_col("{green test:s}")
glue_col("{green test;s}")
glue_col("{green test~s}")
glue_col("{green test^s}")
glue_col("{green test*s}")
glue_col("{green test+s}")
glue_col("<<green tests>>", .open = '<<', .close = '>>')
```

### reprex using `glue_col` doc's example

```r
# using glue_col example

# Prints okay (exact copy)
if (require(crayon)) {
  glue_col("{blue foo bar}")

  glue_col("{blue 1 + 1 = {1 + 1}}")

  white_on_grey <- bgBlack $ white
  glue_col("{white_on_grey
    Roses are {red {colors()[[552]]}}
    Violets are {blue {colors()[[26]]}}
    `glue_col()` can show {red c}{yellow o}{green l}{cyan o}{blue r}{magenta s}
    and {bold bold} and {underline underline} too!
    }")
}

# Does not print (adds a single quote to each fstring)

if (require(crayon)) {
  glue_col("{blue foo bar'}")

  glue_col("{blue 1 + 1' = {1 + 1}}")

  white_on_grey <- bgBlack $ white
  glue_col("{white_on_grey
    Rose's are {red {colors()[[552]]}}
    Violets are {blue {colors()[[26]]}}
    `glue_col()` can show {red c}{yellow o}{green l}{cyan o}{blue r}{magenta s}
    and {bold bold} and {underline underline} too!
    }")
}
```

Crayon appears to not require escaping for colored output

```
library(crayon)
# Works as expected
cat(blue("Hello", "world!\n"))
cat(blue("Hello", "world'!\n"))
```

### LINKED ISSUE #235: Add an option to disable parsing of quotes
as reported in https://github.com/r-lib/cli/issues/370

### PR DESCRIPTION
**Pull request by jimhester**
Fixes #235, fixes #158

### CODE REVIEW

**Comment by jimhester** [comment]
I don't know if `literal` is the best name for this argument, happy for alternative names if you think of something better.

File: R/glue.R [hunk 1]
```diff
@@ -28,6 +28,9 @@
 #'   value is replaced by the value of `.null`.
 #' @param .comment \[`character(1)`: \sQuote{#}]\cr Value to use as the comment
 #'   character.
+#' @param .literal \[`boolean(1)`: \sQuote{FALSE}]\cr If `True` treat the text
```
Comment (jennybc) [inline R/glue.R:31]:
```suggestion
#' @param .literal \[`boolean(1)`: \sQuote{FALSE}]\cr If `TRUE` treat the text
```

**Review by jennybc — COMMENTED** [review]
I'm using this PR to get a little into deeper into glue, more generally.

My first goal is to make sure I totally "get" the problem, as it's quite artificial with straightforward, direct `glue()` usage. Why would anyone put text they want to be treated literally inside `{}`?

I just want to record that your Slack suggestion to ponder `glue_col()` is very fruitful and I think this PR could potentially close #158 as well.

This is pretty compelling. Sorry about the screenshot, but it's the easiest way to quickly look at this color stuff.

Screen Shot 2021-11-19 at 9 30 15 AM

**Comment by jimhester** [comment]
We might want to explicitly put `.literal` in the arguments for `glue_col()`

**Comment by jennybc** [comment]
@jimhester I think we should leave this for the next release, which can be soon. But let's not send this feature out slightly underbaked.

**Pull request merged**